subprojects/core-api/src/main/java/org/gradle/api/tasks/TaskInputs.java

```diff
@@ -88,7 +88,7 @@
      * @param name The name of the property. Must not be null.
      * @param value The value for the property. Can be null.
      */
-    TaskInputs property(String name, @Nullable Object value);
+    TaskInputPropertyBuilder property(String name, @Nullable Object value);
```

Comment:
We did the exact same thing with `TaskInputs.file()` etc. before. We've been issuing deprecation warnings for `inputs.property(...).property()` for a long time now.

Reply:
Not that chaining `inputs.property().property()` (or even `inputs.property().file()` still works, it just issues a deprecation warning.

Reply:
I think we did not issue a deprecation warning for `inputs.property().property()`, since we returned `this` from the method. If my understanding is correct, we now fail the build: https://github.com/gradle/gradle/blob/0feb60e8e04497671db842af863286c3711c3f31/subprojects/core/src/main/java/org/gradle/api/internal/tasks/AbstractTaskInputsDeprecatingTaskPropertyBuilder.java#L32-L34
I guess we cannot do that all of a sudden.

Reply:
Yep, fixed